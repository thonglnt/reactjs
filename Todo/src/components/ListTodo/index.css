.list-todo {
    padding-left: 10px;
    margin-top: 5px;
}