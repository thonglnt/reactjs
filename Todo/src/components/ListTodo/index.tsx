// Lib
import { useState } from 'react';

// Types
import { ITodo } from 'types/todo';

// Components
import TodoItem from '@components/TodoItem';

export type ListTodoProps = {
  todos: ITodo[];
  onRemoveTodo: (id: string) => void;
  onToggleTodo: (id: string) => void;
};

const ListTodo = ({ todos, onRemoveTodo, onToggleTodo }: ListTodoProps) => {
  const [removingTodoId, setRemovingTodoId] = useState<string[] | undefined>([]);

  const handleRemoveTodo = (id: string) => {
    setRemovingTodoId(removingTodoId?.concat(id));
    onRemoveTodo(id);
  };

  return (
    <ul className='list-todo'>
      {todos.map((item) => (
        <TodoItem
          onToggleTodo={onToggleTodo}
          onRemoveTodo={handleRemoveTodo}
          key={item.id}
          id={item.id}
          task={item.task}
          status={item.status}
          isRemoving={removingTodoId?.includes(item.id)}
        />
      ))}
    </ul>
  );
};

export default ListTodo;
