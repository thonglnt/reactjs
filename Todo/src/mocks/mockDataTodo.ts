// types
import { ITodo } from 'types/todo';

export const mockDataTodo: ITodo[] = [
  {
    id: '1',
    task: 'task 1',
    status: true,
    idUser: '1',
  },
  {
    id: '2',
    task: 'task 1',
    status: false,
    idUser: '2',
  },
];
