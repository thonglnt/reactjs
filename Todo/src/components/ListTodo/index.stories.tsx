// Lib
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

// Components
import ListTodo from '@components/ListTodo';
import { mockDataTodo } from '@mocks/mockDataTodo';

storiesOf('ListTodo', module).add('ListTodoSample', () => (
  <ListTodo
    onRemoveTodo={action('onRemoveTodo')}
    onToggleTodo={action('onToggleTodo')}
    todos={mockDataTodo}
  />
));
