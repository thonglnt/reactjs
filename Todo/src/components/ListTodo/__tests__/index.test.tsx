// Lib
import { render, screen, fireEvent } from '@testing-library/react';

// Components
import ListTodo, { ListTodoProps } from '@components/ListTodo';
import { mockDataTodo } from '@mocks/mockDataTodo';

describe('Test ListTodo component', () => {
  const initListTodo: ListTodoProps = {
    todos: mockDataTodo,
    onRemoveTodo: jest.fn(),
    onToggleTodo: jest.fn(),
  };

  test('It should render correctly with initial Props', () => {
    const { container } = render(<ListTodo {...initListTodo} />);

    expect(container).toMatchSnapshot();
  });

  test('It should call remove todo', () => {
    render(<ListTodo {...initListTodo} />);
    fireEvent.click(screen.getAllByTestId('delete-todo')[0]);

    expect(initListTodo.onRemoveTodo).toHaveBeenCalledTimes(1);
  });
});
