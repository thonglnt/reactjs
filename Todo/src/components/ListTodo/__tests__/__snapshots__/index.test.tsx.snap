// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListTodo component It should render correctly with initial Props 1`] = `
<div>
  <ul
    class="list-todo"
  >
    <li
      class="todo-item"
    >
      <div
        class="round"
      >
        <input
          checked=""
          type="checkbox"
        />
        <label
          data-testid="check-done-todo"
          for="checkbox"
        />
      </div>
      <p
        class="todo-task"
      >
        <s>
          task 1
        </s>
      </p>
      <img
        alt="delete"
        class="icon"
        data-testid="delete-todo"
        src="IMAGE_MOCK"
      />
    </li>
    <li
      class="todo-item"
    >
      <div
        class="round"
      >
        <input
          type="checkbox"
        />
        <label
          data-testid="check-done-todo"
          for="checkbox"
        />
      </div>
      <p
        class="todo-task"
      >
        task 1
      </p>
      <img
        alt="delete"
        class="icon"
        data-testid="delete-todo"
        src="IMAGE_MOCK"
      />
    </li>
  </ul>
</div>
`;
